content/en/blog/_posts/2024-07-18-kubernetes-1.31-deprecations-and-removals.md

```diff
@@ -0,0 +1,116 @@
+
+For further details on the cloud provider integrations read our [v1.29 Cloud Provider Integrations feature blog](https://kubernetes.io/blog/2023/12/14/cloud-provider-integration-changes/) and for additional context about the in-tree code removal we invite you to check the ([v1.29 deprecation blog](https://kubernetes.io/blog/2023/11/16/kubernetes-1-29-upcoming-changes/#removal-of-in-tree-integrations-with-cloud-providers-kep-2395-https-kep-k8s-io-2395)).
+
+The latter blog also contains useful information for users who need to migrate to version 1.29 and later.
+
+
+### Removal of Kubelet Flag `--keep-terminated-pod-volumes`
+The Kubelet flag `--keep-terminated-pod-volumes`, which was deprecated in 2017, has now been removed.
+
+You can find more details in the pull request [#122082](https://github.com/kubernetes/kubernetes/pull/122082).
+
+### Removal of CephFS volume plugin 
+[CephFS volume plugin](https://kubernetes.io/docs/concepts/storage/volumes/#cephfs) was removed in this release and the `cephFS` volume type became non-functional. 
+
+Its recommended that you use [CephFS CSI driver](https://github.com/ceph/ceph-csi/) third party storage driver instead. A re-deployment of your application is required to use the new driver if you were using CephFS volume plugin before upgrading cluster version to v1.31.
+
+CephFS volume plugin was formally marked as deprecated in v1.28.
+
+### Removal of Ceph RBD volume plugin
+[Ceph RBD volume plugin](https://kubernetes.io/docs/concepts/storage/storage-classes/#ceph-rbd) was removed in this release together with its CSI migration support and the ceph RBD volume type became non-functional. 
+
+Its recommended that you use use the [RBD CSI driver](https://github.com/ceph/ceph-csi/) in your clusters instead. A re-deployment of your application is required to use the new driver if you were using Ceph RBD volume plugin before upgrading cluster version to v1.31.
+
+CephRBD volume plugin was formally marked as deprecated in v1.28.
+
+### Deprecation of Non-CSI Volume Limit Plugins in Kube-scheduler 
+
+The `kube-scheduler` has deprecated all non-CSI volume limit plugins and removed them from the [default plugins](https://kubernetes.io/docs/reference/scheduling/config/), including:
+
+- AzureDiskLimits
+- CinderLimits
+- EBSLimits
+- GCEPDLimits
+
+Its recommended that you use `NodeVolumeLimits` plugin instead because it can handle the same functionality as the removed plugins since those volume types have been migrated to CSI. Please replace the deprecated plugins with the `NodeVolumeLimits` plugin if you explicitly use them in the [scheduler config](https://kubernetes.io/docs/reference/scheduling/config/). The `AzureDiskLimits`, `CinderLimits`, `EBSLimits`, `GCEPDLimits` plugins will be removed in Kubernetes v1.32.
```

Comment:
```suggestion
Its recommended that you use `NodeVolumeLimits` plugin instead because it can handle the same functionality as the removed plugins since those volume types have been migrated to CSI. Please replace the deprecated plugins with the `NodeVolumeLimits` plugin if you explicitly use them in the [scheduler config](https://kubernetes.io/docs/reference/scheduling/config/). The `AzureDiskLimits`, `CinderLimits`, `EBSLimits`, and `GCEPDLimits` plugins will be removed in Kubernetes v1.32.
```